Automattic/jetpack
# WordPress.com Block Editor Deprecate and move functionality to mu wpcom

## Proposed changes:

* Deprecate WordPress.com Block Editor for self-hosted
* Move the functionality to `jetpack-mu-wpcom` package
* Fixed Unit Tests while moving them

Regarding deprecation notices and removal of the code, I took a different approach to https://github.com/Automattic/jetpack/pull/37260 or https://github.com/Automattic/jetpack/pull/37189 since the jetpack-mu-wpcom package is not available for the jetpack plugin.
The steps I take are as follows. After removing the loading of the file from the [module-extras.php](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/module-extras.php#L45) I decided to mark the file as deprecated as well as the init and the constructor. But leave the file as is since I cannot replace the functionality with the one in the mu-wpcom package due to being out of reach.
### Other information:


## Jetpack product discussion


pfwV0U-9G-p2
## Does this pull request change what data or activity we track or use?


No
## Testing instructions:





* Check in self-hosted and see that Justify and Underline formatting Tools are no longer available.
![image](https://github.com/Automattic/jetpack/assets/13239096/7e3af98a-f966-4ca9-a0b4-a23be76d6c0b)

* Make sure deprecation notices are not present unless the Class is declared explicitly. 
* Check that Simple and WoA still keep the justify and Underline formatting Tools
